server/__test__/service/PostReview.test.js

```diff
@@ -0,0 +1,171 @@
+
+        const err = new Error('Test Error');
+
+        validateFieldSpy.mockReturnValueOnce(true);
+        postReviewSpy.mockRejectedValueOnce(err);
+
+        expect(reviewService.createNewReview(testSuccess)).rejects.toThrow();
+    })
+
+    //test that fails validation
+    test('Fails when Validating Data', async() => {
+        //test fail object
+        const testFailNoUN = {
+            username: "",
+            imageUrl: "test.png",
+            rating: 5,
+            content: "Test Fail"
+        }
+
+        const err = new Error('Test Error');
+
+        validateFieldSpy.mockReturnValueOnce(err);
+        
+        expect(reviewService.createNewReview(testFailNoUN)).rejects.toThrow();
+    })
+
+    /**
+     * Tests for validateFields function
+     */
+
+    //test to see if function succeeds
+    test('Success of Validate Data', () => {
+        //test object
+        const testSuccess = {
+            username: "John",
+            imageUrl: "Doe.png",
+            rating: "5",
+            content: "Test Success"
+        }
+
+        const result = reviewService.validateFields(testSuccess);
+
+        expect(result).toBeTruthy();
+    })
+
+    //test to see if function Fails with empty values
+    test('Failure of Validate Data (Empty Values)', () => {
+        //test fail object
+        const testFailNoUN = {
+            username: "",
+            imageUrl: "test.png",
+            rating: 5,
+            content: "Test Fail"
+        }
+
+        const testFailNoImg = {
+            username: "John",
+            imageUrl: "",
+            rating: "5",
+            content: "Test Fail"
+        }
+        
+        const testFailNoRating = {
+            username: "John",
+            imageUrl: "test.png",
+            rating: "",
+            content: "Test Fail"
+        }
+        
+        const testFailNoContent = {
+            username: "John",
+            imageUrl: "test.png",
+            rating: "2",
+            content: ""
+        }
+
+        expect(() => reviewService.validateFields(testFailNoUN)).toThrow('No Username Present');
+        expect(() => reviewService.validateFields(testFailNoImg)).toThrow('No Image Present');
+        expect(() => reviewService.validateFields(testFailNoRating)).toThrow('No Rating Present');
+        expect(() => reviewService.validateFields(testFailNoContent)).toThrow('No Content Present');
+    })
+
+    //test to see if function fails with incorrect image file
+    test('Failure to Validate Data (Incorrect Image Extension)', () => {
+        const testFailureinExtension = {
+            username: "John",
+            imageUrl: "Doe.psg",
+            rating: "5",
+            content: "Test Success"
+        }
+
+        expect(() => reviewService.validateFields(testFailureinExtension)).toThrow('Image Extension Unapplicable');
+    })
+
+    //test to see if function fails with outside rating range
+    test('Failure to Validate Data (Outside rating range)', () => {
```

Comment:
What happens if the rating is a decimal number between 1 and 5?  We should have the rating as an integer.